src/task_scl_eff_pot_hab.py

```diff
@@ -179,6 +201,93 @@ def poly_export(self, polys, scl_name):
+        )
+        return (
+            threshold_variable_image.addBands(
+                [variable_high_probability, self.zone_image]
+            )
+            .updateMask(self.zone_image)
+            .rename(
+                [
+                    "{}_all_vals".format(variable_name),
+                    "{}_high_prob_vals".format(variable_name),
+                    "zone",
+                ]
+            )
+        )
+
+    def histogram_calc(self, threshold_calc_image, histogram_reducer):
+        return ee.List(
+            ee.Dictionary(
+                threshold_calc_image.reduceRegion(
+                    reducer=histogram_reducer,
+                    geometry=ee.Geometry.Polygon(self.extent),
+                    scale=self.scale,
+                    crs=self.crs,
+                    maxPixels=self.ee_max_pixels,
+                    tileScale=16,
+                )
+            ).get("groups")
+        )
+
+    def zone_threshold_calc(self, zone_histogram_objects):
+        # histogram_format is mapped over zone_histogram_objects
+        # Each zone_object  contains the counts and histograms for a zone
+        def histogram_format(zone_object):
+            zone_dictionay = ee.Dictionary(zone_object)
+            zone_number = zone_dictionay.get("zone")  # currently this does not get used
+            pixel_counts = ee.List(zone_dictionay.get("count"))
+            pixel_count_all = ee.Number(pixel_counts.get(0))
+            pixel_count_high = ee.Number(pixel_counts.get(1))
+            histograms = ee.List(zone_dictionay.get("histogram"))
+
+            bin = ee.Array(histograms.get(0)).slice(1, 0, 1).toList().flatten()
+            histogram_all_pixels = (
+                ee.Array(histograms.get(0)).slice(1, 1, 2).toList().flatten()
+            )
+            histogram_high_probability_pixels = (
+                ee.Array(histograms.get(1)).slice(1, 1, 2).toList().flatten()
+            )
+            histogram_combine = bin.zip(
+                histogram_all_pixels.zip(histogram_high_probability_pixels)
+            )
+
+            # threshold_calcis mapped over hist_combine
+            # each histogram object is a list of list where each list item is [bin_value, frequency_all_pixels, frequency_high_probablity_pixels]
+            def threshold_calc(histogram_object):
+                histogram_item_list = ee.List(histogram_object)
+                bin_value = ee.Number(histogram_item_list.get(0))
+                frequency_values = ee.List(histogram_item_list.get(1))
+                frequency_all = ee.Number(frequency_values.get(0))
+                frequency_high = ee.Number(frequency_values.get(1))
+                proportion_all = frequency_all.divide(pixel_count_all)
+                proportion_high = frequency_high.divide(pixel_count_high)
+                difference = proportion_high.subtract(proportion_all)
+                return ee.Number(bin_value.multiply(difference.gte(0))).int()
```

Comment:
A comment about this line and the next might be helpful... like what would our future selves need to do to enhance this key selection bit? Also -- do we need to protect against the possibility of `difference` never being gte 0?